src/Model/Group.php

```diff
@@ -0,0 +1,430 @@
+		if ($use_gcontact) {
+			$sql = 'SELECT `gcontact`.`id` AS `contact-id` FROM `group_member`
+					INNER JOIN `contact` ON `contact`.`id` = `group_member`.`contact-id`
+					INNER JOIN `gcontact` ON `gcontact`.`nurl` = `contact`.`nurl`
+				WHERE ' . $condition;
+			$param_arr = array_merge([$sql], $group_ids);
+			$stmt = call_user_func_array('dba::p', $param_arr);
+		} else {
+			$condition_array = array_merge([$condition], $group_ids);
+			$stmt = dba::select('group_member', ['contact-id'], $condition_array);
+		}
+
+		$return = array();
+		if (DBM::is_result($stmt)) {
+			while($group_member = dba::fetch($stmt)) {
+				$return[] = $group_member['contact-id'];
+			}
+		}
+
+		if ($check_dead && !$use_gcontact) {
+			require_once 'include/acl_selectors.php';
+			$return = prune_deadguys($return);
+		}
+		return $return;
+	}
+
+	/**
+	 * @brief Returns a templated group selection list
+	 *
+	 * @param int $uid
+	 * @param int $gid An optional pre-selected group
+	 * @param string $label An optional label of the list
+	 * @return string
+	 */
+	public static function displayGroupSelection($uid, $gid = 0, $label = '')
+	{
+		$o = '';
+
+		$groups = dba::select('group', [], ['deleted' => 0, 'uid' => $uid], ['order' => ['name' => 'ASC']]);
+
+		$display_groups = [
+			[
+				'name' => '',
+				'id' => '0',
+				'selected' => ''
+			]
+		];
+		foreach ($groups as $group) {
+			$display_groups[] = [
+				'name' => $group['name'],
+				'id' => $group['id'],
+				'selected' => $gid == $group['id'] ? 'true' : ''
+			];
+		}
+		logger('groups: ' . print_r($display_groups, true));
+
+		if ($label == '') {
+			$label = t('Default privacy group for new contacts');
+		}
+
+		$o = replace_macros(get_markup_template('group_selection.tpl'), array(
+			'$label' => $label,
+			'$groups' => $display_groups
+		));
+		return $o;
+	}
+
+	/**
+	 * @brief Create group sidebar widget
+	 *
+	 * @param string $every
+	 * @param string $each
+	 * @param string $editmode
+	 * 	'standard' => include link 'Edit groups'
+	 * 	'extended' => include link 'Create new group'
+	 * 	'full' => include link 'Create new group' and provide for each group a link to edit this group
+	 * @param int $group_id
+	 * @param int $cid
+	 * @return string
+	 */
+	public static function sidebarWidget($every = 'contacts', $each = 'group', $editmode = 'standard', $group_id = 0, $cid = 0)
+	{
+		$o = '';
+
+		if (!local_user()) {
+			return '';
+		}
+
+		$display_groups = [
+			[
+				'text' => t('Everybody'),
+				'id' => 0,
+				'selected' => (($group_id == 0) ? 'group-selected' : ''),
+				'href' => $every,
+			]
+		];
+
+		$groups = dba::select('group', [], ['deleted' => 0, 'uid' => local_user()], ['order' => ['name' => 'ASC']]);
+
+		$member_of = array();
+		if ($cid) {
+			$member_of = self::getByContactIdForUserId(local_user(), $cid);
+		}
+
+		if (DBM::is_result($groups)) {
+			foreach ($groups as $group) {
```

Comment:
Here is the same like above. Additionally you can omit the ```is_result``` since the ```dba::fetch``` will handle it correctly if there is no result.